### DISCUSSION TITLE
Nextflow keeps throwing a 'Plugin system was already setup' exception for all versions after 22.06.0

### DISCUSSION THREAD

**Discussion by UmmulkiramR** [nextflow-io/nextflow#4019] [Q&A]
## Bug report 

I recently upgraded the version of the nextflow.io library to version **22.10.7**  in our spring boot app and since then my code has started breaking. My app calls the `run()` method of the `K8SDriverLauncher` class to create workflow pods. 
Now, after the upgrade each time I run my workflows it runs perfectly fine for the 1st time but for all the subsequent runs it gives me this - `IllegalArgumentException("Plugin system was already setup")`  

### Expected behavior and actual behavior

Upon further investigation what I found is that as a part of the **22.06.0-edge** release the following 2 changes were introduced in the library's code. 

1. In the K8SDriverLauncher class, in the `loadConfig()` method there is an additional call made to the `Plugins.init() `
2. In the PluginsFacade class the `setup()` method is changed to `init()` and an if statement is added that throws the `IllegalArgumentException`

What is happening is that the `init()` method of the PluginsFacade class looks for the presence of the  plugins manager on this line -> `if( manager )` every time.  For the 1st run it does not find a manager so it creates one. For the 2nd, 3rd, 4th...  runs it finds the manager it created during the 1st run and that's when it throws the exception.  

Here is the code snippet

```
void init(boolean embedded=false) {
        if( manager )
            throw new IllegalArgumentException("Plugin system was already setup")

        log.debug "Setting up plugin manager > mode=${mode}; embedded=$embedded; plugins-dir=$root; core-plugins: ${defaultPlugins.toSortedString()}"
        // make sure plugins dir exists
        if( mode!=DEV_MODE && !FilesEx.mkdirs(root) )
            throw new IOException("Unable to create plugins dir: $root")

        this.manager = createManager(root)
        this.updater = createUpdater(root, manager)
        manager.loadPlugins()
        if( embedded ) {
            manager.startPlugins()
            this.embedded = embedded
        }
    }
```

Why have the  IllegalArgumentException thrown when it does find the plugins manager? What is wrong with one being there already? 

### Program output 

Here's the stack trace of the exception I see.
```
java.lang.IllegalArgumentException: Plugin system was already setup
	at nextflow.plugin.PluginsFacade.init(PluginsFacade.groovy:190) ~[nf-commons-22.10.7.jar!/:na]
	at nextflow.plugin.Plugins.init(Plugins.groovy:41) ~[nf-commons-22.10.7.jar!/:na]
	at nextflow.plugin.Plugins.init(Plugins.groovy) ~[nf-commons-22.10.7.jar!/:na]
	at nextflow.plugin.Plugins$init.call(Unknown Source) ~[na:na]
	at nextflow.k8s.K8sDriverLauncher.loadConfig(K8sDriverLauncher.groovy:268) ~[nextflow-22.10.7.jar!/:na]
	at nextflow.k8s.K8sDriverLauncher.makeConfig(K8sDriverLauncher.groovy:299) ~[nextflow-22.10.7.jar!/:na]
	at nextflow.k8s.K8sDriverLauncher.run(K8sDriverLauncher.groovy:142) ~[nextflow-22.10.7.jar!/:na]
```

### Environment 

* Nextflow version: 22.06.0-edge onward
* Operating system: Linux, MacOs

**Comment by UmmulkiramR** [comment]
Hi Team, is there an update on this?

**Comment by Buwujiu** [comment]
Hi, is there any updates on this? This is blocking our projects, please take a look when you have chance.

**Comment by azulverdosa** [comment]
We’re seeing the same issue here.

**Reply by pditommaso** [reply]
Dear community, open-source is meant to allow everybody to access the code, debug it and contribute back improvements and bug fixes. This is not a support forum.
